platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -3160,6 +3143,22 @@ namespace platoon_strategic_ihp
         // because it is a rough plan, assume vehicle can always reach to the target speed in a lanelet
         maneuver_msg.lane_following_maneuver.end_time = current_time + ros::Duration(config_.time_step);
         maneuver_msg.lane_following_maneuver.lane_ids = { std::to_string(lane_id) };
+
+        ROS_DEBUG_STREAM("in compose maneuver lane id:"<< lane_id);
+
+        lanelet::ConstLanelet current_lanelet = wm_->getMap()->laneletLayer.get(lane_id);
+        if(!wm_->getMapRoutingGraph()->following(current_lanelet, false).empty())
```

Comment:
Since this is ignoring the ending downtrack distance won't this check often result in incorrect lanelet ids?

Reply:
Seems like this method might be more appropriate 
https://github.com/usdot-fhwa-stol/carma-platform/blob/8689cec36bf9df40ea13e53a65b92c3c70ae9238/route_following_plugin/src/route_following_plugin.cpp#L300

But maybe I'm missing the context of this function?

Reply:
Can leave as is since this is limited to platooning and works for ihp2